### DISCUSSION TITLE
[Enhancement] Leverage proper native (hw accelerated) Animations

### DISCUSSION THREAD

**Discussion by GiampaoloGabba** [dotnet/maui#65] [General] [answered]
## Summary
The current Animation implementation do not make use of native platform APIs, rather they animate properties on the elements (by setting up its own timing).

This is good for simple cases but it can became "sluggish" depending on what we are doing.

Furthermore this is a very inefficient method to animate things when the underlying platforms can leverage hardware acceleration!

With the new MAUI we have the opportunity to do things the way they should, so lets create an efficient and performant animation framework!
The old implementation can be kept for platforms that don't support hardware acceleration (i'm just speculating here). But at least for Android and iOS we should take advantage of native animations

## API Changes
If possibile, it would be great to maintain the same API for the users. Then, under the hood, link these methods to Native code that perform an hardware accelerated animation.

## Intended Use Case
Animations are becoming more and more a key part of mobile UX. We need a solid framework that leverage the true power of our hardware, we should not make too much compromise in this area.

**Comment by jsuarezruiz** [comment, marked as answer]
I like the proposal. It is something that we have really considered and discussed internally. This change would be related to the proposed changes in https://github.com/dotnet/maui/issues/17
